docs/style.md

```diff
@@ -539,6 +539,245 @@ import { unstable_Pagination as Pagination } from 'carbon-components-react';
+
+    it('should provide a data-testid attribute on the outermost DOM node', () => {
+      const { container } = render(<ComponentName className="test" />);
+      expect(screen.getByTestId('component-test-id')).toBeInTheDocument();
+      expect(container.firstChild).toHaveAttribute('class', 'test');
+    });
+
+    it('should place the `className` prop on the outermost DOM node', () => {
+      const { container } = render(<ComponentName className="test" />);
+      expect(container.firstChild).toHaveAttribute('class', 'test');
+    });
+
+    it('should place extra props on the outermost DOM node', () => {
+      const { container } = render(<ComponentName data-testid="test" />);
+      expect(container.firstChild).toHaveAttribute('data-testid', 'test');
+    });
+
+    // id
+    // -----
+    // When a component accepts an id prop, it's important
+    // that the node on which the id is placed is consistent
+    // between minor versions. As a result, tests that you
+    // write for id should make assertions around id being
+    // placed on the same node.
+    id(
+      'should place the `id` prop on the same DOM node between minor versions',
+      () => {
+        const { container } = render(<ComponentName data-testid="test" />);
+        expect(container.firstChild).toHaveAttribute('id', 'test');
+      }
+    );
+
+    // Optional ref tests
+    // A component that accepts a ref falls in one of three scenarios:
+    // 1. A class component
+    // 2. A component that uses React.forwardRef and placed
+    //    it on an HTML element
+    // 3. A component that uses React.forwardRef and uses
+    //    useImperativeHandle to decorate the ref (this is
+    //    uncommon but can come up)
+  });
+});
+```
+
+##### `ComponentName-test.a11y.js`
+
+- Use `accessibility-checker` and `axe`
+- Optionally configure common props to ensure component variants do not contain
+  accessibility errors.
+
+```js
+describe('ComponentName - AVT1', () => {
+  it('should have no aXe violations', async () => {
+    render(<ComponentName />);
+    await expect(
+      screen.getByLabelText('or a similar query')
+    ).toHaveNoAxeViolations();
+  });
+
+  it('should have no AC violations', async () => {
+    render(<ComponentName />);
+    await expect(
+      screen.getByLabelText('or a similar query')
+    ).toHaveNoACViolations('ComponentName');
+  });
+});
+```
+
+##### `ComponentName-test.e2e.js`
+
+- These tests are ran with Cypress, which is a bit different in syntax and
+  mental model than Jest/JSDOM tests. It's recommended to become familiar with
+  the core concepts by reading the
+  [Cypress introduction documentation](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress).
+- Trigger Events with ``via`cy.real\*();`
+- Keyboard navigation/interaction testing is best done here, in a simulated
+  browser environment, not JSDOM.
+- Take snapshots with Percy
+  - Percy snapshots should be used very sparingly (1-2 per component)
```

Comment:
Sort of. It's dependent on how we config and what we want to cover. IIRC we'll have a quota of 200k snaps a month? My thought was that it would run on every PR.

- ~55 components @ 2 snaps each
  - 110 snaps per run
- Ideally we'd check chrome/safari/firefox
  - 110 snaps * 3 browsers = 330 snaps per run
- If we run this test suite in PR checks it would be ran on every update w/ latest from `main` (average 2 of those? 3?)
  - 330 snaps * 3 updates on every PR = 990 snaps per PR
- Average past two months PR count is ~100 pull requests per month
  - 990 snaps per PR * 100 PRs a month = 99,000

Lots of variables there too. Add one more snap per component and it's 150k, 4 snaps per component and we hit quota.

An alternative option would be to not run VRT on every PR, but instead only once a week through a GH action and have it automatically open defect issue(s) for us to address before we release.

TLDR - We'll probably have to choose between more snaps ran less frequently, or less snaps ran more frequently

Reply:
When we last had Percy enabled, it only tested Firefox and Chrome, so that would reduce it a little bit